File: WitchcraftApp/src/test/java/com/tarbadev/witchcraft/TspToCupConverterTest.java
package com.tarbadev.witchcraft;

import org.junit.Before;
import org.junit.Test;

import static org.assertj.core.api.Java6Assertions.assertThat;

public class TspToCupConverterTest {
  private TspToCupConverter subject;

  @Before
  public void setUp() {
    subject = new TspToCupConverter();
  }

  @Test
  public void convert() {
    Double tsp = 48.0;
    Double cup = 1.0;

    assertThat(subject.convert(tsp)).isEqualTo(cup);

    tsp = 3.0;
    cup = 0.0625;

    assertThat(subject.convert(tsp)).isEqualTo(cup);
  }
}

File: WitchcraftApp/src/test/java/com/tarbadev/witchcraft/converter/TspToCupConverterTest.kt
package com.tarbadev.witchcraft.converter

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class TspToCupConverterTest {
    private lateinit var tspToCupConverter: TspToCupConverter

    @BeforeEach
    fun setUp() {
        tspToCupConverter = TspToCupConverter()
    }

    @Test
    fun convert() {
        var tsp = 48.0
        var cup = 1.0

        assertEquals(cup, tspToCupConverter.convert(tsp))

        tsp = 3.0
        cup = 0.0625

        assertEquals(cup, tspToCupConverter.convert(tsp))
    }
}
